File: src/main/java/org/voegtle/weatherwidget/diagram/LeoDiagramActivity.java
package org.voegtle.weatherwidget.diagram;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import org.voegtle.weatherwidget.R;
import org.voegtle.weatherwidget.util.StringUtil;

public class LeoDiagramActivity extends DiagramActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    addDiagram(DiagramEnum.leo_regen);
    addDiagram(DiagramEnum.leo_wind);
    addDiagram(DiagramEnum.leo_solar);
    addDiagram(DiagramEnum.leo_solar_average);
    addDiagram(DiagramEnum.leo_solar_production);
    addDiagram(DiagramEnum.leo_paderborn);
    addDiagram(DiagramEnum.leo_lastyear);
    Intent intent = getIntent();
    if (intent != null) {
      String title = intent.getStringExtra(LeoDiagramActivity.class.getName());
      if (StringUtil.isNotEmpty(title)) {
        setTitle(title);
      }
    }
  }

  @Override
  public boolean onCreateOptionsMenu(Menu menu) {
    MenuInflater inflater = getMenuInflater();
    inflater.inflate(R.menu.leo_diagram_activity_menu, menu);
    return super.onCreateOptionsMenu(menu);
  }

  @Override
  protected boolean onCustomItemSelected(MenuItem item) {
    switch (item.getItemId()) {
      case R.id.action_leo_rain:
        viewPager.setCurrentItem(0, true);
        return true;
      case R.id.action_leo_wind:
        viewPager.setCurrentItem(1, true);
        return true;
      case R.id.action_leo_solar:
        viewPager.setCurrentItem(2, true);
        return true;
      case R.id.action_solar_daily:
        viewPager.setCurrentItem(3, true);
        break;
      case R.id.action_solar_monthly:
        viewPager.setCurrentItem(4, true);
        break;
      case R.id.action_leo_paderborn:
        viewPager.setCurrentItem(5, true);
        return true;
      case R.id.action_leo_last_year:
        viewPager.setCurrentItem(6, true);
        return true;
    }
    return false;
  }
}

File: src/main/java/org/voegtle/weatherwidget/diagram/LeoDiagramActivity.kt
package org.voegtle.weatherwidget.diagram

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import org.voegtle.weatherwidget.R
import org.voegtle.weatherwidget.util.StringUtil

class LeoDiagramActivity : DiagramActivity() {
  override val placeHolderId: Int? = R.drawable.station_leopoldshoehe

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    addDiagram(DiagramEnum.leo_regen)
    addDiagram(DiagramEnum.leo_wind)
    addDiagram(DiagramEnum.leo_power)
    addDiagram(DiagramEnum.leo_solar_average)
    addDiagram(DiagramEnum.leo_solar_production)
    addDiagram(DiagramEnum.leo_30days)
    addDiagram(DiagramEnum.leo_lastyear)
  }

  override fun onCreateOptionsMenu(menu: Menu): Boolean {
    val inflater = menuInflater
    inflater.inflate(R.menu.leo_diagram_activity_menu, menu)
    return super.onCreateOptionsMenu(menu)
  }

  override fun onCustomItemSelected(item: MenuItem): Boolean =
      when (item.itemId) {
        R.id.action_leo_rain -> updateViewPager(0)
        R.id.action_leo_wind -> updateViewPager(1)
        R.id.action_leo_solar -> updateViewPager(2)
        R.id.action_solar_daily -> updateViewPager(3)
        R.id.action_solar_monthly -> updateViewPager(4)
        R.id.action_leo_30days -> updateViewPager(5)
        R.id.action_leo_last_year -> updateViewPager(6)
        else -> false
      }
}
